testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java

```diff
@@ -0,0 +1,837 @@
+        // TODO: support redirects from Iceberg to Hive
+        assertQueryFailure(() -> onTrino().executeQuery("TABLE " + icebergTableName))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Not an Iceberg table: default." + tableName);
+//        assertResultsEqual(
+//                onTrino().executeQuery("TABLE " + icebergTableName),
+//                onTrino().executeQuery("TABLE " + tableName)); // unqualified
+
+        onTrino().executeQuery("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testIcebergToPartitionedHivePartitionsRedirect()
+    {
+        String tableName = "hive_partitioned_table_" + randomTableSuffix();
+        String hiveTableName = "hive.default." + tableName;
+
+        createHiveTable(hiveTableName, true);
+
+        // TODO: support redirects from Iceberg to Hive
+        assertQueryFailure(() -> onTrino().executeQuery("TABLE iceberg.default.\"" + tableName + "$partitions\""))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Not an Iceberg table: default." + tableName);
+
+        onTrino().executeQuery("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testIcebergToUnpartitionedHivePartitionsRedirect()
+    {
+        String tableName = "hive_unpartitioned_table_" + randomTableSuffix();
+        String hiveTableName = "hive.default." + tableName;
+
+        createHiveTable(hiveTableName, false);
+
+        // TODO: support redirects from Iceberg to Hive
+        assertQueryFailure(() -> onTrino().executeQuery("TABLE iceberg.default.\"" + tableName + "$partitions\""))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Not an Iceberg table: default." + tableName);
+
+        onTrino().executeQuery("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "schemaAndPartitioning")
+    public void testIcebergToHiveInsert(String schema, boolean partitioned)
+    {
+        String tableName = "hive_insert_by_iceberg_" + randomTableSuffix();
+        String hiveTableName = "hive." + schema + "." + tableName;
+        String icebergTableName = "iceberg." + schema + "." + tableName;
+
+        createHiveTable(hiveTableName, partitioned, false);
+
+        // TODO: support redirects from Iceberg to Hive
+        assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO " + icebergTableName + " VALUES (6, false, -17), (7, true, 1)"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Not an Iceberg table: " + schema + "." + tableName);
+
+        onTrino().executeQuery("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "schemaAndPartitioning")
+    public void testHiveToIcebergInsert(String schema, boolean partitioned)
```

Comment:
nit: move above `testIcebergToHiveInsert` (so we always have same flavor of test ordered hive->incberg first icebarg->hive later)

Reply:
> so we always have same flavor of test ordered hive->incberg first icebarg->hive later


i tried to make it so, but actually it's not as ordered (initial part of the class uses different ordering than neighbourhood here).

i can reorder. but i also though about splitting the class into two (hive->iceberg redirects in `hive` test package and iceberg->hive redirects in `iceberg` package). wdyt?

Reply:
Splitting is better idea IMO. If you can structure it so not too much code is repeated

Reply:
discussed offline, and decide to split.